feature/home/src/main/java/io/filmtime/feature/home/HomeScreen.kt

```diff
@@ -34,11 +50,46 @@ fun HomeScreen(
   onMovieClick: (tmdbId: Int) -> Unit,
   onShowClick: (tmdbId: Int) -> Unit,
 ) {
+
+
   val state by viewModel.state.collectAsStateWithLifecycle()
+  var load by remember { mutableStateOf(false) }
+
+
+  LaunchedEffect(load) {
+    if (load) {
+      viewModel.retry()
+      viewModel.loadTrendingMovies()
+      viewModel.loadTrendingShows()
+      load = false
+
+    }
+  }
 
+  val context = LocalContext.current
   if (state.isLoading) {
+
     LoadingVideoSectionRow(numberOfSections = 2)
-  } else {
+  }else if (state.error != null){
+    when( state.error){
+      is GeneralError.NetworkError->{
+        NetworkErroScreen(context){
+          load = true
```

Comment:
based on previous comment here we only need to call the retry function of view model